### DISCUSSION TITLE
compile error

### DISCUSSION THREAD

**Discussion by tlzfighting** [contiki-ng/contiki-ng#2319] [Q&A] [answered]
I wrote a demo myself, and its file directory is as follows:
                               
![图片](https://user-images.githubusercontent.com/95647449/206266117-737b506b-25a5-4b9e-b74e-96aa4cedeed5.png)
The contents of the makefile file are as follows:
![图片](https://user-images.githubusercontent.com/95647449/206266276-be6acc6b-1a29-492d-9a3d-963ed671010b.png)
I get an error when I compile：
![图片](https://user-images.githubusercontent.com/95647449/206266910-061a757e-1f92-47c8-bfd4-32ed975924cb.png)
Please tell me how to solve it. Thank you for your valuable advice！

**Comment by pjonsson** [comment, marked as answer]
The general form of make commands is `make <variables> output-file`, so `make TARGET=cc2538dk dht11` should be closer to what you want. The Contiki-NG build system is not completely straight forward, but a make tutorial might clarify some of the general principles.

There should not be a need to use `sudo` with make for Contiki-NG unless the documentation explicitly mentions it. I can see that if the permissions are wrong, it might be required. Outside the container, what does `id -u` output?
